lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,468 @@
+        new_cube = check_cube_coordinates(
+            prob_lightning_cube, new_cube)
+        return new_cube
+
+    def apply_ice(self, prob_lightning_cube, ice_cube):
+        """
+        Modify Nowcast of lightning probability with ice data from a radar
+        composite (VII; Vertically Integrated Ice)
+
+        Args:
+            prob_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability.
+                The forecast_period coord is modified in-place to "minutes".
+            ice_cube (iris.cube.Cube):
+                Analysis of vertically integrated ice (VII) from radar
+                thresholded at 0.5, 1.0, 2.0 kg m^-2
+                Units of threshold coord modified in-place to kg m^-2
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing updated nowcast lightning probability.
+                This cube will have the same dimensions and meta-data as
+                prob_lightning_cube.
+                The influence of the data in ice_cube reduces linearly to zero
+                as forecast_period increases to 2H30M.
+
+        Raises:
+            iris.exceptions.ConstraintMismatchError:
+                If ice_cube does not contain the expected thresholds.
+        """
+        prob_lightning_cube.coord('forecast_period').convert_units('minutes')
+        # check prob-ice threshold units are as expected
+        ice_cube.coord('threshold').convert_units('kg m^-2')
+        new_cube_list = iris.cube.CubeList([])
+        err_string = "No matching prob(Ice) cube for threshold {}"
+        for cube_slice in prob_lightning_cube.slices_over('time'):
+            fcmins = cube_slice.coord('forecast_period').points[0]
+            for threshold, prob_max in zip(self.ice_thresholds,
+                                           self.ice_scaling):
+                ice_slice = ice_cube.extract(
+                    iris.Constraint(
+                        threshold=lambda t: isclose(t.point, threshold)))
+                if not isinstance(ice_slice, iris.cube.Cube):
+                    raise ConstraintMismatchError(err_string.format(threshold))
+                # Linearly reduce impact of ice as fcmins increases to 2H30M.
+                ice_scaling = [0., (prob_max * (1. - (fcmins / 150.)))]
+                cube_slice.data = np.maximum(
+                    rescale(ice_slice.data,
+                            data_range=(0., 1.),
+                            scale_range=ice_scaling,
+                            clip=True),
+                    cube_slice.data)
+            new_cube_list.append(cube_slice)
+
+        new_cube = new_cube_list.merge_cube()
+        new_cube = check_cube_coordinates(
+            prob_lightning_cube, new_cube)
+        return new_cube
+
+    def process(self, cubelist):
+        """
+        Produce Nowcast of lightning probability
+
+        Args:
+            cubelist (iris.cube.CubeList):
```

Comment:
This is just a style-thing. The separation of the cubelist could happen at the CLI level instead. I think this method was used because at one point there were a large number of arguments which have now been moved elsewhere.